### ISSUE TITLE
ArgoCD complains "repoURL" parameter is missing

### ISSUE THREAD

**Issue by rogerxaic** [oboukili/terraform-provider-argocd#97]
Hi all, 

I am trying to deploy an ArgoCD `Application` with this Terraform provider but I'm getting an error. I have been able to successfully deploy a `Repository` and a `Project`.

I have a terraform resource such as this:

```terraform
resource "argocd_application" "application" {

  metadata {
    name      = "somename"
    namespace = "somenamespace"
  }

  spec {
    project = "someproject"

    source {
      repo_url = "https://server/project.git"
    }

    [...]

  }
}
```

This gives me the following error on apply:

```logs
2021/09/09 19:19:20 Terraform apply | Plan: 1 to add, 0 to change, 0 to destroy.
 2021/09/09 19:19:23 Terraform apply | argocd_application.application: Creating...
 2021/09/09 19:19:25 Terraform apply | 
 2021/09/09 19:19:25 Terraform apply | Error: rpc error: code = InvalidArgument desc = application spec is invalid: InvalidSpecError: spec.source.repoURL and spec.source.path either spec.source.chart are required
 2021/09/09 19:19:25 Terraform apply | 
 2021/09/09 19:19:25 Terraform apply |   with argocd_application.application,
 2021/09/09 19:19:25 Terraform apply |   on redacted.tf line 7, in resource "argocd_application" "application":
 2021/09/09 19:19:25 Terraform apply |    7: resource "argocd_application" "application" {
 2021/09/09 19:19:25 Terraform apply | 
 2021/09/09 19:19:25 Terraform APPLY error: Terraform APPLY errorexit status 1
 2021/09/09 19:19:25 Could not execute action
```

Apparently, ArgoCD expects a `repoURL` parameter, but this provider doesn't expect it. That means that if I add it, Terraform will complain I've got an extra unexpected parameter.

I'm using ArgoCD v2.0.5.

Any idea how to solve this problem?
Thanks in advance

**Comment by oboukili** [comment]
Hello, you are missing either spec.source.path or spec.source.chart, you can read the examples in the documentation.

**Comment by rogerxaic** [comment]
Hello @oboukili, thanks for your quick answer!

I looked in the documentation and I interpreted that I only needed "path" if I had put my files in another directory than root (i.e., I thought `path`'s default value would be `.`). Indeed, the documentation says `path` is _Optional_.

```
The source block has the following attributes:

    repo_url - (Required) string, repository URL of the application manifests.
    path - (Optional) string, directory path within the Git repository.
    [...]
```

With the help you provided, it is now working, thanks!



I would suggest changing the invalid InvalidSpecError from `spec.source.repoURL and spec.source.path either spec.source.chart are required` to `spec.source.repoURL and either spec.source.path or spec.source.chart are required` but I guess this is a response from ArgoCD so not under your control.

Is it possible to modify the ArgoCD provider so that it checks for either one of these values? That would allow you to give a customized error message on `terraform plan` instead of `terraform apply`.
